packages/telemetry/src/make-slog-sender.js

```diff
@@ -1,32 +1,121 @@
+  const { env = {}, stateDir: stateDirOption, ...otherOpts } = opts;
+  const { SLOGSENDER = DEFAULT_SLOGSENDER_MODULE, ...otherEnv } = env;
+
+  const slogSenderModules = SLOGSENDER.split(',').map(modulePath =>
+    modulePath.startsWith('.')
+      ? // Resolve relative to the current working directory.
+        path.resolve(modulePath)
+      : modulePath,
+  );
+  if (
+    otherEnv.SLOGFILE &&
+    !slogSenderModules.includes(SLOGFILE_SENDER_MODULE)
+  ) {
+    slogSenderModules.push(SLOGFILE_SENDER_MODULE);
   }
 
-  if (slogSenderModule.startsWith('.')) {
-    // Resolve relative to the current working directory.
-    slogSenderModule = path.resolve(slogSenderModule);
+  if (!slogSenderModules.length) {
+    return undefined;
   }
 
-  console.warn(`Loading makeSlogSender from ${slogSenderModule}`);
-  const { makeSlogSender: maker } = await import(slogSenderModule);
-  if (typeof maker !== 'function') {
-    throw Error(`${slogSenderModule} did not export a makeSlogSender function`);
+  console.warn(
+    `Loading slog sender${
+      slogSenderModules.length > 1 ? 's' : ''
+    } from ${slogSenderModules.join(', ')}`,
+  );
+
+  const makersInfo = await Promise.all(
+    slogSenderModules.map(async moduleIdentifier =>
+      import(moduleIdentifier)
+        .then(
+          /** @param {{makeSlogSender: (opts: {}) => Promise<SlogSender | undefined>}} module */ ({
+            makeSlogSender: maker,
+          }) =>
+            typeof maker === 'function'
+              ? { maker, moduleIdentifier }
+              : Promise.reject(
+                  new Error(`No 'makeSlogSender' function exported by module`),
+                ),
+        )
+        .catch(err => {
+          console.warn(
+            `Failed to load slog sender from ${moduleIdentifier}.`,
+            err,
+          );
+          return undefined;
+        }),
+    ),
+  );
+
+  if (!makersInfo.filter(Boolean).length) {
+    return undefined;
   }
 
-  const { stateDir = tmp.dirSync().name } = makerOpts;
-  if (stateDir !== makerOpts.stateDir) {
+  let stateDir = stateDirOption;
+
+  if (stateDir === undefined) {
+    stateDir = tmp.dirSync().name;
     console.warn(`Using ${stateDir} for stateDir`);
   }
-  return maker({ ...makerOpts, stateDir });
+
+  const senders = await Promise.all(
+    makersInfo.map(async makerInfo => {
+      if (!makerInfo || makerInfo.maker === makeSlogSender) {
+        return undefined;
+      }
+      const { maker, moduleIdentifier } = makerInfo;
+      return maker({
+        ...otherOpts,
+        stateDir,
+        env: { SLOGSENDER: moduleIdentifier, ...otherEnv },
+      });
+    }),
+  ).then(
+    potentialSenders =>
+      /** @type {SlogSender[]} */ (potentialSenders.filter(Boolean)),
+  );
+
+  if (!senders.length) {
+    return undefined;
+  } else if (senders.length === 1) {
+    return senders[0];
+  } else {
+    const hasSenderNeedingJsonObj = senders.some(sender => sender.length === 2);
```

Comment:
Oh no!  An arity check?!!

This deserves a `FIXME` comment explaining why the dodginess is worth it, and what other (more explicit) mechanisms could replace it.  And also `sender.length >= 2`.